pkg/client/events_query/client.go

```diff
@@ -0,0 +1,273 @@
+package eventsquery
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"sync"
+
+	"go.uber.org/multierr"
+
+	"pocket/pkg/client"
+	"pocket/pkg/either"
+	"pocket/pkg/observable"
+	"pocket/pkg/observable/channel"
+)
+
+const requestIdFmt = "request_%d"
+
+var _ client.EventsQueryClient = (*eventsQueryClient)(nil)
+
+// TODO_CONSIDERATION: the cosmos-sdk CLI code seems to use a cometbft RPC client
+// which includes a `#EventsBytes()` method for a similar purpose. Perhaps we could
```

Comment:
Unfortunately, I was not able to get theirs to work. :confused: 

It's complete speculation on my part at this point but I do recall the behavior expressed when trying to use their was identical to a point where I was debugging ours and it was using binary websocket messages instead of text messages. The symptom was basically that it could connect and wouldn't complain when sending requests, it just would never send anything back.

Reply:
FWIW, I kinda like the relative simplicity of ours. Regardless, I think we have good interfaces now to adapt to whatever. :rocket:

Reply:
> Regardless, I think we have good interfaces now to adapt to whatever. 🚀

Yea, as much as I'd prefer to use theirs and not build our own, I think this is the sort of primitie that will enable us to extend to any RPC service w/o dependency on Cosmos 👌 .

One ask: Link to this comment thread in the code.